src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -569,6 +536,7 @@ private function addTableToSchema(Schema $schema): void
         $table->addIndex(['queue_name']);
         $table->addIndex(['available_at']);
         $table->addIndex(['delivered_at']);
+        $table->addIndex(['queue_name', 'available_at', 'delivered_at', 'id']);
```

Comment:
@psihius are you sure the ID field is needed?

The select query doesn't use the ID field in the where condition.